pkg/controller/s3/bucket.go

```diff
@@ -98,43 +98,45 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	}
 
 	if _, err := e.s3client.HeadBucketRequest(&awss3.HeadBucketInput{Bucket: aws.String(meta.GetExternalName(cr))}).Send(ctx); err != nil {
-		return managed.ExternalObservation{}, awsclient.Wrap(resource.Ignore(s3.IsNotFound, err), errHead)
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(s3.IsNotFound, err), errHead)
 	}
 
-	cr.Status.AtProvider = s3.GenerateBucketObservation(meta.GetExternalName(cr))
+	cr.Status.AtProvider = s3.GenerateBucketObservation(cr)
+
+	lateInit := false
 
-	current := cr.Spec.ForProvider.DeepCopy()
 	for _, awsClient := range e.subresourceClients {
-		err := awsClient.LateInitialize(ctx, cr)
-		if err != nil {
-			return managed.ExternalObservation{ResourceExists: true, ResourceUpToDate: false}, err
+		if awsClient.SubresourceExists(cr) {
+			err := awsClient.LateInitialize(ctx, cr)
+			if err != nil {
+				return managed.ExternalObservation{ResourceExists: true, ResourceUpToDate: false}, err
+			}
+			lateInit = true
```

Comment:
The `ResourceLateInitialized` field in return variable triggers a spec update which has to delete the status subresource, so we treat carefully and not call it unless there is a change in spec we need to save. In this instance, it seems like subresource existence causes `ResourceLateInitialized` to be returned as `true`, which might render status subresource useless. I'd suggest returning `ResourceLateInitialized: true` only if spec has changed.

Reply:
Hmm, so could one potential change to this could include returning a boolean value from the `LateInitialize` which reflects if some changes have been made? If that condition is met, then we return `ResourceLateInitialized: true`?